app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
```

Comment:
The foreign key is improperly implemented. As per the given ERD, the Post association should have Post belong to User through Author, an alias of User, like below:
```
# app/models/post.rb
belongs_to :author, class_name: 'User'
has_many :comments
has_many :likes
```
the `foreign_key: :author_id` is not necessary. 

Kindly ensure to improve upon this aspect.